worker-core/src/main/java/com/hpe/caf/worker/core/WorkerTaskImpl.java

```diff
@@ -143,11 +143,26 @@ private TaskMessage createResponseMessage(final boolean includeTaskContext, fina
 
         final String responseMessageType = response.getMessageType();
 
+        //  Check if the 'trackTo' field needs to be reset.
+        final TrackingInfo trackingInfo;
+        if (response.getResetTrackTo()) {
+            //  Reset trackTo field to null;
+            final TrackingInfo taskMessageTracking = taskMessage.getTracking();
+            trackingInfo = new TrackingInfo(taskMessageTracking.getJobTaskId(),
+                    taskMessageTracking.getStatusCheckTime(),
+                    taskMessageTracking.getStatusCheckUrl(),
+                    taskMessageTracking.getTrackingPipe(),
+                    null);
```

Comment:
- Don't copy all the fields - instead use the copy constructor and then call the `setTrackTo()` mutator to change that one field.  For an example see the `updateTaskId()` method.
- You call `taskMessage.getTracking()` and then use it without checking if it is `null`.  I suspect this will cause a `NullPointerException` for messages that haven't been submitted using the Job Service.